llama-index-integrations/embeddings/llama-index-embeddings-nvidia/llama_index/embeddings/nvidia/base.py

```diff
@@ -0,0 +1,207 @@
+"""NVIDIA embeddings file."""
+
+from typing import Any, List, Literal, Optional
+
+from llama_index.core.base.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+)
+from llama_index.core.bridge.pydantic import Field, PrivateAttr, BaseModel
+from llama_index.core.callbacks.base import CallbackManager
+from llama_index.core.base.llms.generic_utils import get_from_param_or_env
+
+from openai import OpenAI, AsyncOpenAI
+
+BASE_RETRIEVAL_URL = "https://ai.api.nvidia.com/v1/retrieval/nvidia"
+DEFAULT_MODEL = "NV-Embed-QA"
+
+
+class Model(BaseModel):
+    id: str
+
+
+class NVIDIAEmbedding(BaseEmbedding):
```

Comment:
Is this another case where it's worth sub-classing OpenAIEmbedding?

Reply:
this may be possible in the future. right now there are some quirks in the api spec, e.g. input_type / truncate / dimensions, and behavior, e.g. max batch size, which make it simpler to use the client and not subclass.

Reply:
- agree for truncate, it's additional.
- dimension is actually an openai parameter that isn't supported on the nim endpoints, so it would have to be subtracted, which looks possible so long as it is kept None.
- concerning is input_type, which additional but conditioned on the method, e.g. query embedding needs it to be "query" while text embedding needs it to be "passage".

what would you suggest for input_type?

Reply:
Ah input type is a good point -- no easy way to provide that without subclassing the text and query embedding methods.

Lets keep it as is then 👍🏻